AWSKinesis/AWSAbstractKinesisRecorder.m

```diff
@@ -25,9 +25,18 @@
 NSUInteger const AWSKinesisAbstractClientByteLimitDefault = 5 * 1024 * 1024; // 5MB
 NSTimeInterval const AWSKinesisAbstractClientAgeLimitDefault = 0.0; // Keeps the data indefinitely unless it hits the size limit.
 NSString *const AWSKinesisAbstractClientUserAgent = @"recorder";
-NSUInteger const AWSKinesisAbstractClientBatchRecordByteLimitDefault = 512 * 1024 * 1024;
+NSUInteger const AWSKinesisAbstractClientBatchRecordByteLimitDefault = 512 * 1024; // 512KB
+// The max size of records that Kinesis streams today allows is 5MB. Do not change
+// the constant below unless that fact has changed.
+NSUInteger const AWSKinesisAbstractClientMaxSizeForPutRecords = 5 * 1024 * 1024;
```

Comment:
This change is now over 2 years old. The code might have changed quite a bit. I have simply forked the repo and made the necessary changes to be used in our Company because of the lack of enthusiasm here.

I shall try to address the above comments nevertheless.